pil-analyzer/src/evaluator.rs

```diff
@@ -655,31 +762,35 @@ mod internal {
         Ok(match b {
             BuiltinFunction::ArrayLen => match arguments.pop().unwrap().as_ref() {
                 Value::Array(arr) => Value::Integer((arr.len() as u64).into()),
-                v => Err(EvalError::TypeError(format!(
+                v => panic!(
                     "Expected array for std::array::len, but got {v}: {}",
                     v.type_name()
-                )))?,
+                ),
             },
             BuiltinFunction::Panic => {
                 let msg = match arguments.pop().unwrap().as_ref() {
                     Value::String(msg) => msg.clone(),
-                    // As long as we do not yet have types, we just format any argument.
-                    x => x.to_string(),
+                    v => panic!(
```

Comment:
Shouldn't this return an `EvalError::TypeError`?

Reply:
See above.